core/src/test/scala/quasar/plugin/postgres/datasource/PostgresDatasourceSpec.scala

```diff
@@ -194,44 +196,158 @@ object PostgresDatasourceSpec
     }
 
     "path to extant non-empty table returns rows as line-delimited json" >>* {
-      val path =
-        ResourcePath.root() / ResourceName("pgsrcSchemaA") / ResourceName("widgets")
-
       val widgets = List(
         Widget("X349", 34.23, 12.5),
         Widget("XY34", 64.25, 23.1),
         Widget("CR12", 0.023, 40.33))
 
-      val setup = for {
-        _ <- sql"""DROP TABLE IF EXISTS "pgsrcSchemaA"."widgets"""".update.run
-        _ <- sql"""CREATE TABLE "pgsrcSchemaA"."widgets" (serial VARCHAR, width DOUBLE PRECISION, height DOUBLE PRECISION)""".update.run
-        load = """INSERT INTO "pgsrcSchemaA"."widgets" (serial, width, height) VALUES (?, ?, ?)"""
-        _ <- Update[Widget](load).updateMany(widgets)
-      } yield ()
+      for {
+        _ <- loadWidgets(widgets)
+        (_, ws) <- resultsOf[Widget](widgetsRead())
+      } yield ws must containTheSameElementsAs(widgets)
+    }
+  }
+
+  "mask pushdown" >> {
+    val maskWidgets = List(
+      Widget("A", 1.1, 2.2),
+      Widget("B", 3.3, 4.4),
+      Widget("C", 5.5, 6.6),
+      Widget("D", 7.7, 8.8))
+
+    "projects when all masked paths begin with a field" >>* {
+      val mask: ScalarStage = ScalarStage.Mask(Map(
+        CPath.parse(".serial") -> Set(ColumnType.String),
+        CPath.parse(".height") -> Set(ColumnType.Number)))
+
+      val expected = List(
+        Json("serial" := "A", "height" := 2.2),
+        Json("serial" := "B", "height" := 4.4),
+        Json("serial" := "C", "height" := 6.6),
+        Json("serial" := "D", "height" := 8.8))
 
       for {
-        _ <- setup.transact(xa)
+        _ <- loadWidgets(maskWidgets)
+        (_, ws) <- resultsOf[Json](widgetsRead(ScalarStages(IdStatus.ExcludeId, List(mask))))
+      } yield ws must containTheSameElementsAs(expected)
+    }
 
-        qr <- datasource.evaluate(InterpretedRead(path, ScalarStages.Id))
+    "noop when masked paths being with an index" >>* {
```

Comment:
postgres can support mask pushdown for indices, just not indices at the start of the path?

Reply:
This implementation doesn't support any index masking, the example should probably read something like "noop when a masked path doesn't begin with a field".

Reply:
I'll change it right quick